### PR TITLE
Refactors crew records

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
I have attempted or otherwise started this project at least 4 times. I am sick of it being on my calendar. The code needs it. I need it.

- This makes crew records a proper datum rather than assigning properties record.fields.
- General, medical, and security records are merged.
- Did some slight refactoring here and there for things that looked obvious.
- Wanted states are now defined (and you can suspect someone through sechud)
- pAI (unrelated but annoying) had some poorly named exported types that i made more specific
- Job icons are moved back to the JS side (I wanted to get icons for initial rank without passing trim)

<details>
<summary>previews</summary>

Editable fields & security console
CM6d74brnC

Medical records
bFJErsvOaN

Look and feel of the more current version
cxGruQsJpP

</details>

## Why It's Good For The Game
TGUI'd some of the worst UIs in the game.
Creating new records is made much simpler. 
Manifest_inject is made readable.
Probably bug fixes

## Changelog

:cl:
refactor: Crew records have been refactored.
refactor: Medical records -> TGUI
refactor: Security records -> TGUI
refactor: Warrants console -> TGUI
qol: Players are now alerted when their fines are paid off.
qol: Cleaned up sec hud examination text.
qol: Adding and deleting crimes is easier.
qol: Writing crimes in the console sets players to arrest.
qol: You can now mark someone as a suspect through sec hud.
/:cl:

### CODE REVIEW

File: code/datums/records/record.dm [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	cdi_d = "No diseases",
+	alg = "None",
+	alg_d = "No allergies",
+	medical_notes = "No notes.",
+	medical_notes_d = "No notes.",
+	p_stat = "Active",
+	m_stat = "Stable",
+	criminal = "None",
+	citation = list(),
+	crim = list(),
```
Comment (ShizCalev) [inline code/datums/records/record.dm:124]: please fucking kill `crim`.

File: code/datums/records/record.dm [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	mi_dis_d = "No disabilities have been diagnosed at the moment.",
+	ma_dis = "None",
+	ma_dis_d = "No disabilities have been diagnosed at the moment.",
+	cdi = "None",
+	cdi_d = "No diseases",
+	alg = "None",
+	alg_d = "No allergies",
+	medical_notes = "No notes.",
+	medical_notes_d = "No notes.",
+	p_stat = "Active",
```
Comment (ShizCalev) [inline code/datums/records/record.dm:120]: what the fuck is `p_stat`

File: code/datums/records/record.dm [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	m_stat = "Stable",
```
Comment (ShizCalev) [inline code/datums/records/record.dm:121]: what the fuck is `m_stat`.

File: code/datums/records/record.dm [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+	trim = "Unassigned",
+	initial_rank = "Unassigned",
+	age = 18,
+	species = "Human",
+	gender = "Other",
+	character_appearance,
+	blood_type = "?",
+	dna = "Unknown",
+	fingerprint = "?????",
+	mi_dis = "None",
```
Comment (ShizCalev) [inline code/datums/records/record.dm:110]: `mi_dis`

File: code/datums/records/record.dm [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+	mi_dis_d = "No disabilities have been diagnosed at the moment.",
+	ma_dis = "None",
```
Comment (ShizCalev) [inline code/datums/records/record.dm:112]: `ma_dis`

File: code/datums/records/record.dm [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+	var/criminal
+	/// Citations list
+	var/citation = list()
+	/// Crimes list
+	var/crim = list()
+	/// Security notes
+	var/security_notes
+
+/datum/record/crew/New(
+	id = "000000",
```
Comment (ShizCalev) [inline code/datums/records/record.dm:98]: `id_number`

**Review by ShizCalev — CHANGES_REQUESTED** [review]
these are really fucking terribly vague and are better off with ACTUALLY HUMAN READABLE NAMES.

**Comment by jlsnow301** [comment]
@ShizCalev these awful var names are on the chopping block for sure. just getting it sorted before i start mass renaming vars

**Comment by jlsnow301** [comment]
just going to redo the security, medical, records consoles idc

File: code/datums/records/record.dm (hunk 4 above)
Comment (timothymtorres) [reply code/datums/records/record.dm:109]: minor disability

File: code/datums/records/record.dm (hunk 5 above)
Comment (timothymtorres) [reply code/datums/records/record.dm:111]: major disability

**Comment by timothymtorres** [comment]
You should post the gifs you shared on the discord to the OP.  Also you are doing god's work man!

**Comment by jlsnow301** [comment]
Should be good for TM / review

**Comment by TheSmallBlue** [comment]
A very common complaint I hear in game that I 100% agree with: Why take away the ability to search by name/job on both consoles? It makes everything way harder than it has to be...

**Comment by jlsnow301** [comment]
> A very common complaint I hear in game that I 100% agree with: Why take away the ability to search by name/job on both consoles? It makes everything way harder than it has to be...

jobs search is now in. in addition to the icons, added earlier. you can now search job/name/dna or job/name/fingerprint respective of console

**Comment by jlsnow301** [comment]
Ready. Reviews appreciated

**Comment by ArthuroPep** [comment]
SEChud seems to not work properly now

**Comment by OverwatchVoice** [comment]
Some errors / omissions I noticed:

- Attempting to interface with the security records console as an AI throws an NTOS crash.
- Adding notes to crew in medical records doesn't show who made that note, as far as I can tell.

**Comment by jlsnow301** [comment]
> * Attempting to interface with the security records console as an AI throws an NTOS crash.
> * Adding notes to crew in medical records doesn't show who made that note, as far as I can tell.
> * No way to add / remove / edit security / medical record entries

The first two, fixed.
The last: Not sure about yet. Especially adding new records: How will I get images? 
Though I may reimplement editing and deleting, the problem is that they're linked and directly affect one another.
Medical records have no login && users could simply delete security records (faster than ever)

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
Problems I have:

1. security record consoles log out automatically if someone without access opens it
2. medical records don't need to log in to see people's medical records, only the notes
3. people's information is not editable (age, species, gender, DNA, job), this is a big problem because it ruins things like adding people to the manifest.
4. there's no way to remove people from the manifest, there used to have a mass delete button to delete all records too. Removing people from the manifest is important because silicon with crew-based lawsets would be countered by removing people from the manifest.
5. Setting a crime on someone will set them to arrest but the UI will still say they aren't

**Comment by jlsnow301** [comment]
> Problems I have:
> 
> 1. security record consoles log out automatically if someone without access opens it
> 2. medical records don't need to log in to see people's medical records, only the notes
> 3. people's information is not editable (age, species, gender, DNA, job), this is a big problem because it ruins things like adding people to the manifest.
> 4. there's no way to remove people from the manifest, there used to have a mass delete button to delete all records too. Removing people from the manifest is important because silicon with crew-based lawsets would be countered by removing people from the manifest.
> 5. Setting a crime on someone will set them to arrest but the UI will still say they aren't

outside of adding entirely new records (confused about this), you can now purge and delete specific records, also add information to currently existing ones (all fields text editable)

**Comment by jlsnow301** [comment]
You can now create new records by uploading a photo. This is ready

**Review by tralezab — APPROVED** [review]
We'll do it live! (It's now documented, and I didn't notice anything outrageously wrong)

**Comment by JupiterJaeden** [comment]
I would ask three things: First, that you remove the new character limit on crime names; it's way too short. Second, that you please remove the "quality of life" change of automatically setting suspects to arrest. There are several reasons you might not want someone set to arrest automatically when you put in a crime; perhaps you only want them set to suspected, perhaps you are filling it in retroactively after already having arrested them. 

And finally, PLEASE get the bugs under control. Particularly the rampant visual glitches with SecHUDs; sometimes it's either not showing arrest status at all, or sometimes it IS showing arrest status; except when it's set to "none", in which case the last status set beforehand is still visible (i.e. they still appear to be "arrest" even though it is now set to "none"). This second bug seems to be the most recent one.

**Comment by jlsnow301** [comment]
the name portion is intentional, literally why have a description section which is 1024 characters when you could just type the name out of the entire crime. the box says "BRIEF DESCRIPTION".  

the name has the chance to clip on the rapsheet printout and the collapsibles of multiple UIs

please make an issue report for the other stuff

**Comment by jlsnow301** [comment]
wb vandalism

**Comment by JupiterJaeden** [comment]
My point on the automatically setting to arrest thing is that it's not really a quality of life change. It just adds something you have to remember whenever you touch the security records to avoid accidentally setting someone to arrest; which I've already seen happen.

**Pull request merged**